test/Microsoft.ML.Benchmarks/RffTransform.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Runtime.Tools;
+using Microsoft.ML.Trainers.Online;
+using Microsoft.ML.Transforms.Projections;
+using System.IO;
+
+namespace Microsoft.ML.Benchmarks
+{
+    public class RffTransformTrain
+    {
+        private string _dataPath_Digits;
+
+        [GlobalSetup]
+        public void SetupTrainingSpeedTests()
+        {
+            _dataPath_Digits = Path.GetFullPath(TestDatasets.Digits.trainFilename);
+
+            if (!File.Exists(_dataPath_Digits))
+                throw new FileNotFoundException(string.Format(Errors.DatasetNotFound, _dataPath_Digits));
+        }
+
+        [Benchmark]
+        public void CV_Multiclass_Digits_RffTransform_OVAAveragedPerceptron()
+        {
+            string cmd = @"CV k=5 data={" + _dataPath_Digits + "}" +
```

Comment:
I think we should really start writing these tests using the actual API.

Reply:
@justinormont can you help converting this to using actual apis ?

Reply:
it will be something like:

 string dataFile = "digits.csv";

 var mlContext = new MLContext();

 var reader = mlContext.Data.CreateTextReader(
                columns: new[]
                    {
                        new TextLoader.Column("Label", DataKind.R4, 64),
                        new TextLoader.Column("Features", DataKind.R4, 1)
                    },
                hasHeader: true, 
                separator: ','
            );

 var data = reader.Read(dataFile);

var pipeline = mlContext.Transforms.Projection.CreateRandomFourierFeatures("Features", "FeaturesRFF")
                          .Append( mlContext.Transforms.Concatenate("Features", "FeaturesRFF"))
                          .Append(mlContext.MultiClassClassificaiton.Trainers.OneVersusAll( ml.BinaryClassification.Trainers.AveragedPerceptron(numIterations:10)));


var cvResults = mlContext.MulticlassClassification.CrossValidate(data, pipeline , numFolds: 5);









---
In reply to: [240422606](https://github.com/dotnet/machinelearning/pull/1855#discussion_r240422606) [](ancestors = 240422606)

Reply:
One mistake above is that the TextLoader Features column should be for a range of columns 0-63. Not just column 1

Reply:
Also I seem to recall there being some need for caching, at some level of this? E.g., cache checkpoint?